### DISCUSSION TITLE
Enhance navigation-agnostic design

### DISCUSSION THREAD

**Discussion by digitalethics** [johnfactotum/foliate#750] [Ideas]
Currently, Foliate is already pretty agnostic in terms of what navigation a user chooses but it is not fully there yet. It would be nice if you could make it even more agnostic and check current (and when implementing new settings and features) against this design philosophy (if you agree with it). What I have in mind are four ways of navigating Foliate:

- mouse-only (MATE, Xfce, Plasma, etc.)
- keyboard-only (GNOME philosophy)
- touch-only (tablet-use)
- voice-controlled (via Mycroft?)

**Comment by johnfactotum** [comment]
What specific features or issues do you have in mind? A "design philosophy" is pretty vague and not actionable.

**Comment by digitalethics** [comment]
> What specific features or issues do you have in mind? A "design philosophy" is pretty vague and not actionable.

Sorry, I should have been more specific. As this is probably a longer list of interconnected issues, I am going to list actionable items here in this discussion rather than as separate issues to prevent clogging up the issue tracker.

**Comment by digitalethics** [comment]
Some ideas for actionable items to improve **keyboard-only navigation**: 

- [ ] #368 Add keyboard shortcuts to tooltips (easier transition into keyboard-only navigation)
- [ ] Rename Menu -> _Keyboard Shortcuts_ to _Navigation_ (makes less assumptions on how Foliate is going to be used, yet clear enough to include touchpad gestures)
- [ ] In _Keyboard Shortcuts_, Move the <kbd>1</kbd><kbd>2</kbd> navigation from the bottom to the top and replace it with tabs named <kbd>Keyboard Shortcuts</kbd> and <kbd>Touchpad Gestures</kbd> 
- [ ] Allow switching between <kbd>Keyboard Shortcuts</kbd> and <kbd>Touchpad Gestures</kbd> through <kbd><-</kbd> and <kbd>-></kbd> directly rather than having to press <kbd><-/-></kbd>+<kbd>Enter</kbd> as is currently the case with toggling <kbd>1</kbd> and <kbd>2</kbd> (keyboard navigation economics, the less key presses to prompt an action, the better)
- [ ] Add global option to default to an active <kbd>Keyboard Shortcuts</kbd> tab when using Foliate on the _desktop_, and default to the <kbd>Touchpad Gestures</kbd> tab when using Foliate on a _tablet_. _Tablet mode_ (touch-enhanced) or _desktop mode_ could be functionally similar to a global dark mode and include a specific but different default setup optimized per device type and screen resolution (e.g. choice of fonts, spacing, etc. for improved readability out of the box) 
- [ ] When sidebar is active, make <kbd>⇥Tab</kbd> switches to ebook window more visually distinct, so that the user knows that pressing <kbd>⇥Tab</kbd> brings main text window into focus
- [ ] Use <kbd>F9</kbd> to toggle the _Menu_
- [ ] Use <kbd>F10</kbd> to toggle _Find_ field. (<kbd>F10</kbd> is used on Thinkpads for search, not sure whether there is some kind of consensus between Dell, Lenovo, HP on this; maybe there is even a distro-agnostic consensus in the making but I doubt it)
- [ ] Use <kbd>F11</kbd> to toggle between one, two or four column-view (I am biased towards Thinkpads as the <kbd>F11</kbd> key literally suggests columns and would be a perfect match for this functionality, see picture below)
- [ ] Use <kbd>F12</kbd> for toggling fullscreen mode
- [ ] Add a button on the main bar to toggle between light (standard) and dark mode. Aesthetically, I would go with something similar to the night light feature in GNOME or the half-moon symbol commonly shown on websites. The dark mode is too important to be nested in menus and you currently need 5 mouse clicks (at maximum) and even more key presses to toggle it (keyboard navigation economics)

Thinkpad keyboard layout: 
![thinkpad_keyboardlayout](https://user-images.githubusercontent.com/57445091/82434385-15830700-9a93-11ea-9b7b-601997748afb.png)

**Comment by johnfactotum** [comment]
The shortcuts window is a standard GTK widget. The "Keyboard Shortcuts" label in the menu is also standardized across all GNOME apps. F9 for side panel, F10 for menu, and F11 for fullscreen are pretty universal across the vast majority of GTK and many other apps. So these will probably not be changed.

As for toggling dark mode, yes, I agree that currently it's inconvenient if you need to change between themes frequently.
